src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -64,25 +64,54 @@ public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?
     }
 
     /**
-     * Creates a new {@link DockerFromFingerprintFacet} and adds a run. Or adds to an existing one.
+     * Creates a new {@link DockerAncestorFingerprintFacet} and {@link DockerDescendantFingerprintFacet} and adds a run.
+     * Or adds to existing facets.
+     * @param ancestorImageId the ID of the image specified in a {@code FROM} instruction, or null in case of {@code scratch} (i.e., the descendant is a base image)
+     * @param descendantImageId the ID of the image which was built
```

Comment:
`descendantImageId` the "descendant" part is redundant and potentially confusing.
if you have a descendant and an ancestor then the questions that comes to mind is who is in the descendant's parent and who is the ancestors child; (that is - it implies something is in the middle  A -> B -> C where A is the ancestor, B is the thing who you are inferring relationships about and C is the descendant).

Like I said - NIT.